tempodb/encoding/common/interfaces.go

```diff
@@ -61,8 +61,14 @@ func DefaultSearchOptionsWithMaxBytes(maxBytes int) SearchOptions {
 	}
 }
 
+type CompactionRound interface {
```

Comment:
This interface is the "command" sent to the block encodings in the `Compact` function.  It replaces `blocks []*backend.BlockMeta`.  Creating `tempodb/compaction` would mean the dependency chain `tempodb` -> `tempodb/encoding/...` -> `tempodb/compaction`.   The current location seems better since it is single direction.